src/components/dialogs/custom-mui-dialog.tsx

```diff
@@ -9,7 +10,21 @@ import Button from '@mui/material/Button';
 import makeStyles from '@mui/styles/makeStyles';
 import { SubmitButton } from '@gridsuite/commons-ui';
 
-const useStyles = makeStyles((theme) => ({
+interface ICustomMuiDialog {
+    open: boolean;
+    formSchema: any;
+    formMethods: any;
+    onClose: (event: React.MouseEvent) => void;
+    onSave: (data: any) => void;
+    onValidationError?: (errors: any) => void;
+    titleId: string;
+    disabledSave: boolean;
+    removeOptional: boolean;
```

Comment:
```suggestion
    removeOptional?: boolean;
```

should be optional with false by default